### PR TITLE
fix project connectiuon

### PR DESCRIPTION
**Pull request by SaraVieira**
## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [ ] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: packages/prisma-client/src/types.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
+import type { Asset as DbAsset, Project as BaseProject } from "@prisma/client";
 export { Location } from "@prisma/client";
-export type { InstanceProps, Project, User, Breakpoints } from "@prisma/client";
+export type { InstanceProps, User, Breakpoints } from "@prisma/client";
 
 export type Asset = DbAsset & {
   path: string;
   status?: "uploading" | "uploaded";
 };
+export type Project = BaseProject & { Asset: DbAsset[] };
```
Comment (kof) [inline packages/prisma-client/src/types.ts:9]: that type is probably automatically provided by prisma, no? also its not Asset, its assets now

Comment (SaraVieira) [reply packages/prisma-client/src/types.ts]: this has been fixed

File: apps/designer/app/shared/db/project.server.ts [hunk 2]
```diff
@@ -6,41 +6,55 @@ import {
...
+  const newAssets =
+    "assets" in project
+      ? project.assets.map((asset) => ({
+          ...asset,
+          path: getAssetPath(asset),
+        }))
+      : {};
   return {
     ...project,
+    ...newAssets,
```
Comment (Andarist) [inline apps/designer/app/shared/db/project.server.ts:35]: shouldnt this be something like this?
```suggestion
    assets: newAssets,
```

Comment (Andarist) [reply apps/designer/app/shared/db/project.server.ts]: Or rather actually - you probably have meant to create `{ assets: projects.assets.map(() => {...}) }` a couple of lines ago

Comment (SaraVieira) [reply apps/designer/app/shared/db/project.server.ts]: They are both the same idea, just committed your first one :)

File: apps/designer/app/shared/db/project.server.ts [hunk 3]
```diff
@@ -6,41 +6,55 @@ import {
...
 const TreeHistorySchema = z.array(z.string());
 
 export type Project = Omit<BaseProject, "prodTreeIdHistory"> & {
   prodTreeIdHistory: z.infer<typeof TreeHistorySchema>;
 };
 
-const parseProject = (project: BaseProject | null): Project | null => {
+const parseProject = (
+  project: BaseProject | null
+): Project | Omit<BaseProject, "assets"> | null => {
```
Comment (Andarist) [inline apps/designer/app/shared/db/project.server.ts:22]: This return type looks a little bit fishy - I would expect a unified type to come up from this parsing function (so probably smth like `Project | null`).

File: packages/prisma-client/src/types.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
-import type { Asset as DbAsset } from "@prisma/client";
+import type { Asset as DbAsset, Project as BaseProject } from "@prisma/client";
```
Comment (Andarist) [inline packages/prisma-client/src/types.ts:1]: It's a little bit confusing to me that things like `BaseProject` are pointing to different types in different files. It makes it harder to figure out the relationships between those things.

File: apps/designer/app/shared/db/project.server.ts (hunk 2 above)
Comment (Andarist) [reply apps/designer/app/shared/db/project.server.ts]: In here there is a subtle difference and that's also why I've incidentally suggested that first patch at first 😅 It's on the verge of a typo. The problem is that with this patch your `newAssets` are still either an array or an object, the other suggested patch ensures that we spread an object with an optional `assets` property.

File: packages/prisma-client/src/types.ts (hunk 4 above)
Comment (kof) [reply packages/prisma-client/src/types.ts:1]: half-migrated state, we have extracted prisma client into a separate package and now need to move all the augmented types to where they are are used, to the vertical/feature code

File: apps/designer/app/shared/db/project.server.ts [hunk 5]
```diff
@@ -8,39 +8,66 @@ import {
   Project as BaseProject,
 } from "@webstudio-is/prisma-client";
 import * as db from ".";
+import { getAssetPath } from "@webstudio-is/asset-uploader/src/helpers/get-asset-path";
```
Comment (kof) [inline apps/designer/app/shared/db/project.server.ts:11]: this was probably auto-added

File: packages/prisma-client/src/types.ts [hunk 6]
```diff
@@ -1,9 +1,10 @@
...
 export { Location } from "@prisma/client";
-export type { InstanceProps, Project, User, Breakpoints } from "@prisma/client";
+export type { InstanceProps, User, Breakpoints } from "@prisma/client";
 
 export type Asset = DbAsset & {
-  path: string;
+  path?: string;
   status?: "uploading" | "uploaded";
 };
+export type Project = BaseProject & { assets?: Asset[] };
```
Comment (kof) [inline packages/prisma-client/src/types.ts:9]: Isn't prisma supposed to know that assets are part of the Project type and give it to us automatically?

File: apps/designer/app/designer/features/sidebar-left/sidebar-left.tsx [hunk 7]
```diff
@@ -78,7 +78,7 @@ export const SidebarLeft = ({ publish, assets }: SidebarLeftProps) => {
           }}
         >
           <TabContent
-            assets={assets}
+            assets={assets || []}
```
Comment (kof) [inline apps/designer/app/designer/features/sidebar-left/sidebar-left.tsx:81]: should probably be a default inside tab content

**Comment by kof** [comment]
I was just thinknig: do we actually need assets loaded upfront when designer loads? Images on canvas have a URL already built-in, no? 

If we only need images when user opens images tab, should we not fetch them then? I just imagined if a project has hundreds of images, we would fetch a substantial payload even if user doesn't go into images tab.

We could even think of using the remix's router for fetching each tab's data and scripts, so we don't load even the logic for handling contents of each tab. I think long-term there will be quite a lot of logic in there.

**Comment by rpominov** [comment]
I see this locally:

Screenshot 2022-08-19 at 11 24 37

Probably because we changed `schema.prisma`, but didn't create a migration file. I'll make a PR to fix this.

**Comment by kof** [comment]
@SaraVieira said there is no need for migration because changes were on models only, how did this even work on dev?

**Comment by kof** [comment]
Just checked, preview has failed as well

**Comment by rpominov** [comment]
Sorry, no migration is needed actually. I just needed to run `yarn build:prisma` locally.

**Comment by rpominov** [comment]
> Just checked, preview has failed as well

Hm is `yarn build:prisma` run during deployment of a preview?

**Comment by kof** [comment]
in vercel config we run `yarn prisma migrate deploy && yarn build`

**Comment by kof** [comment]
yarn build:prisma inside prisma-client package calls prisma generate and we don't do this on CI, the question is why and how did we survive this before?  Was this broken as we moved prisma client to a package?

**Comment by kof** [comment]
@SaraVieira @andrasbacsai what was the reason we called it  build:prisma instead of "build" inside the prisma-client package? Otherwise it would be running automatically on ci and when we pull we have yo run one simple command from the root.

I remember there was something about it ....

**Comment by andrasbacsai** [comment]
I do not remember. I try to fix it.

**Pull request merged**